portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +28,121 @@ function addRandomGreeting() {
+
+//   // The fetch() function returns a Promise because the request is asynchronous.
+//   const responsePromise = fetch('/data');
+
+//  // When the request is complete, pass the response into handleResponse().
+//   responsePromise.then(response);
+// }
+// function response(response){
+//     console.log(response.text()); 
+//     response.text().then((text) => { document.getElementById("data").innerText = text;})
+// }
+function getRandomQuote() {
+  console.log('Fetching a random quote.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addQuoteToDom);
+}
+
+/** Adds a random quote to the DOM. */
+function addQuoteToDom(quote) {
+  console.log('Adding quote to dom: ' + quote);
+
+  const quoteContainer = document.getElementById('data');
+  quoteContainer.innerText = quote;
+}
+function getDataServer(){
+    fetch('/data').then(response=>response.json()).then((quote)=> {
+        const statsListElement = document.getElementById('data-container');
+        statsListElement.innerHTML= '';
+        statsListElement.appendChild(
+            createListElement('Name:'+quote.name));
+        statsListElement.appendChild(
+            createListElement('institution:'+quote.institution));
+        statsListElement.appendChild(
+            createListElement('nationality:'+quote.nationality));
+
+    });
+    
+}
+function createListElement(text){
+    const liElement = document.createElement('li');
+    liElement.innerText = text;
+    return liElement;
+}
+
+
+function showSlides(n) {
+  var i;
+  var slides = document.getElementsByClassName("mySlides");
+  var dots = document.getElementsByClassName("dot");
+  if (n > slides.length) {slideIndex = 1}
+  if (n < 1) {slideIndex = slides.length}
+  for (i = 0; i < slides.length; i++) {
+      slides[i].style.display = "none";
+  }
+  for (i = 0; i < dots.length; i++) {
+      dots[i].className = dots[i].className.replace(" active", "");
+  }
+  slides[slideIndex-1].style.display = "block";
+  dots[slideIndex-1].className += " active";
+}
+ function openPage(pageName, elmnt, color) {
+    // Hide all elements with class="tabcontent" by default */
+    var i, tabcontent, tablinks;
+    tabcontent = document.getElementsByClassName("tabcontent");
+    for (i = 0; i < tabcontent.length; i++) {
+      tabcontent[i].style.display = "none";
```

Comment:
fyi: This is fine to leave as it will work great - but a more common practice I see is to create a CSS class called something like "hidden" and add or remove that from the element. This also makes it a bit simpler to open one by defautl.